app/email/routes.py

```diff
@@ -0,0 +1,90 @@
+from app.subscribe.store_subscription_data import check_email
+from flask import request, jsonify
+
+from app.email import bp
+from app.models import Users
+from app.errors.errors import (
+    AlreadyExistsError,
+    DatabaseError,
+    InvalidUsageError,
+    UnauthorizedError,
+)
+from flask_cors import cross_origin
+from flask_jwt_extended import jwt_required
+from flask_jwt_extended import current_user
+
+from app import auto, db
+
+
+@bp.route("/email", methods=["GET"])
+@cross_origin()
+@jwt_required()
+@auto.doc()
+def current_email():
+    """
+    Returns the current email address of a logged in user or standard JWT errors if token not available or expired.
+    """
+
+    response = {"currentEmail": current_user.user_email}
+
+    return jsonify(response), 200
+
+
+@bp.route("/email", methods=["PUT"])
+@cross_origin()
+@jwt_required()
+@auto.doc()
+def update_email():
+    """
+    Updates the user's email address.
+
+    Returns an error if:
+    - the user is not logged in or token has expired (standard JWT errors)
+    - any parameters are missing from the request
+    - the new email address is not in a valid format
+    - the new email address does match the address given in the confirmed email address field
+    - the new email address is already in the db
+    """
+
+    request_body = request.get_json(force=True, silent=True)
+
+    for param in ("newEmail", "confirmEmail", "password"):
+        if param not in request_body:
+            raise InvalidUsageError(
+                message=f"{param} must be included in the request body."
+            )
+
+    new_email = request_body["newEmail"]
+    confirm_email = request_body["confirmEmail"]
+    password = request_body["password"]
+
+    if not check_email(new_email):
+        raise InvalidUsageError(
+            message="Cannot update email. Email is not formatted correctly."
+        )
+
+    if new_email != confirm_email:
+        raise InvalidUsageError(
+            message="Cannot update email. New email address and confirm new email address do not match."
+        )
+
+    email_already_used = Users.query.filter_by(user_email=new_email).first()

```

Comment:
I think it makes sense here just to leave this as user = Users.query...

As you know, the object returned is a user. It's easy to understand
```
if user:
    raise AlreadyExistsError(message="A user is already registered with this email.")
```

In fact, in the registration endpoint I used this format (Note that we have a method built in the user class to handle this query):

```
    user = Users.find_by_username(r["email"])
    if user:
        raise UnauthorizedError(message="Email already registered")
    else:
        user = add_user_to_db(
            r["firstName"], r["lastName"], r["email"], r["password"], r["quizId"]
        )
```

It might make sense to rename the find_by_username to find_by_email, but it doesn't matter much to me since they are equivalent at the moment, so I'll leave that up to your choice.

Also, I think you meant to return the email they typed? Right now you are returning a literal message "Cannot update email. Email"

Additionally, this is a place we should be using one_or_none instead of first since only one user should exist per email. If you use the find_by_username function, then this is already set up this way.

Reply:
Discussed and agreed alignment strategy to use class method and AlreadyExistsError.